test/optests_failures_dict.json

```diff
@@ -2,6 +2,32 @@
   "_description": "This is a dict containing failures for tests autogenerated by generate_opcheck_tests. For more details, please see https://docs.google.com/document/d/1Pj5HRZvdOq3xpFpbEjUZp2hBovhy7Wnxw14m6lF2154/edit",
   "_version": 1,
   "data": {
+    "torchvision::ps_roi_align": {
+      "TestPSRoIAlign.test_aot_dispatch_dynamic__test_mps_error_inputs": {
+        "comment": "RuntimeError: MPS does not support ps_roi_align backward with float16 inputs",
+        "status": "xfail"
+      },
+      "TestPSRoIAlign.test_autograd_registration__test_backward[True-mps-0]": {
+        "comment": "NotImplementedError: autograd_registration_check: NYI devices other than CPU/CUDA, got {'mps'}",
+        "status": "xfail"
```

Comment:
CC @zou3519 

I just opened https://github.com/pytorch/pytorch/pull/111797 which I believe could be a fix for the problem I'm facing here:

This test (and a bunch of others)

https://github.com/pytorch/vision/blob/3fb88b3ef1ee8107df74ca776cb57931fe3e9e1e/test/test_ops.py#L186-L189

is parametrized over cpu, CUDA and MPS, and fails on MPS.

I'm trying to `xfail` the MPS parametrization as above (lines 10-12), but `optests` complains with:


> E   RuntimeError: In failures dict, got test name 'TestPSRoIAlign.test_autograd_registration__test_backward[True-mps-0]'. We parsed this as running test 'test_autograd_registration' on 'test_backward[True-mps-0]', but test_backward[True-mps-0] does not exist on the TestCase 'TestPSRoIAlign]. Maybe you need to change the test name?



The problem is, if I replace  `TestPSRoIAlign.test_autograd_registration__test_backward[True-mps-0]` with `TestPSRoIAlign.test_autograd_registration__test_backward` in line 10, then I'm getting and "unexpected success":

> E   torch.testing._internal.optests.generate_tests.OpCheckError: generate_opcheck_tests: Unexpected success for operator torchvision::ps_roi_align on test TestPSRoIAlign.test_autograd_registration__test_backward. This may mean that you have fixed this test failure. Please rerun the test with PYTORCH_OPCHECK_ACCEPT=1 to automatically update the test runner or manually remove the expected failure in the failure dict at /home/nicolashug/dev/vision/test/optests_failures_dict.jsonFor more details, see https://docs.google.com/document/d/1Pj5HRZvdOq3xpFpbEjUZp2hBovhy7Wnxw14m6lF2154/edit

Reply:
@NicolasHug can the tests be run under unittest, or are they pytest only?